language/compiler/ir-to-bytecode/src/context.rs

```diff
@@ -328,6 +348,48 @@ impl<'a> Context<'a> {
         ))
     }
 
+    /// Get the field handle index for the alias, adds it if missing.
+    pub fn field_handle_index(&mut self, handle: FieldHandle) -> Result<FieldHandleIndex> {
+        Ok(FieldHandleIndex(get_or_add_item(
+            &mut self.field_handles,
+            handle,
+        )?))
+    }
+
+    /// Get the struct instantiation index for the alias, adds it if missing.
+    pub fn struct_instantiation_index(
+        &mut self,
+        struct_inst: StructDefInstantiation,
```

Comment:
It looks like every caller to this is constructing the `StructDefInstantiation` before calling. Maybe just take `def` and `type_parameters` as arguments?